crates/fuel-streams-core/src/blocks/subjects.rs

```diff
@@ -10,16 +10,13 @@ pub struct BlocksSubject {
     pub height: Option<BlockHeight>,
 }
 
-impl From<Block> for BlocksSubject {
-    fn from(block: Block) -> Self {
-        BlocksSubject::new().with_height(Some(block.into()))
-    }
-}
-
 impl From<&Block> for BlocksSubject {
     fn from(block: &Block) -> Self {
         let block_height = *block.header().height();
-        BlocksSubject::new().with_height(Some(BlockHeight::from(block_height)))
+        BlocksSubject::new()
+        // TODO: Use correct block producer here
+            .with_producer(Some("0x".to_string()))
```

Comment:
Question: I never got where the block producer is supposed to come - is this something we can extract form the Block or is it external data that needs to passed in here ?

Reply:
I haven't investigated deeply on this yet but I intend deferring for a follow-up PR to avoid delaying this one any longer. Of course, I am open to any ideas. 🙂

Reply:
Thats fine, was just a question as I am not clear on it either :) Lets keep the conversation so we can revisit it later in another MR.